Kirill method

cboettig opened the pull request:
Implements the method proposed by @krlmlr in #7.  Kirill, would be great if you could take a quick glance at this and let me know if this looks like what you proposed.  Does seem to work nicely in my (unit) tests.

krlmlr commented on R/ark.R:
```diff
@@ -36,9 +40,11 @@
                 compress = c("bzip2", "gzip", "xz", "none"),
-                tables = list_tables(db_con)){
+                tables = list_tables(db_con),
+                use_alternate = FALSE){
```
Perhaps a `method = c("keep-open", "window", "sql-window", "manual-window")` would give more flexibility in the long run? I'm not in love with `"keep-open"`, but let's think of a better name for the method than `kirill` ;-)

cboettig replied:
nice, I like this.  I struggled with concise meaningful names and clearly didn't come up with something.  `keep-open` is intuitive.  

I think you're suggesting here that "window" would be the method that uses `OFFSET`, and "sql-window" would be the method that uses `BETWEEN`?  (might be good to bipass my 'automatic' `has_between()` method.

krlmlr commented on R/ark.R:
```diff
@@ -57,39 +64,87 @@ list_tables <- function(db){
+  ## Create header to avoid duplicate column names
+  query <- paste("SELECT * FROM", tablename, "LIMIT 0")
+  header <- DBI::dbGetQuery(db_con, query)
+  readr::write_tsv(header, con, append = FALSE)
```
I wonder if we can get rid of this `dbGetQuery()` call and handle also the initialization in the loop.

cboettig replied:
That would be great.  Any suggestions?  (I tried just commenting this out, which kinda works with the `readr` functions since they actually don't mind the headers being repeated, `readr::read_tsv()` detects this and remove it, but clearly not a robust solution).

krlmlr commented on R/ark.R:
```diff
@@ -99,32 +154,37 @@ ark_chunk <- function(db_con, tablename, start = 1,
 }
 
+## need to convert large integers to characters
+sql_integer <- function(x){
```
Would that be an option?

``` r
sprintf("%.0f", 1e13)
#> [1] "10000000000000"
```

> Created on 2018-07-25 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

cboettig replied:
yes! that's much nicer than messing with scipen option...

cboettig commented:
@krlmlr Thanks for the feedback, I've implemented the `method = ` strategy you proposed, and fixed the handling of sci notation etc.  Merging now as I think this is a solid improvement on the previous approach, but further feedback still welcome.
